pagopa/io-app
# feat: [IOCOM-1391] Flag to enable Messages Home with the new DS

## Short description
This PR adds a persisted preference to display the Messages Home with the new DS.
|Profile|New WiP Home|
|-|-|
|![Simulator Screenshot - iPhone 15 - 2024-05-13 at 16 39 15](https://github.com/pagopa/io-app/assets/5150343/26f68038-0696-4cde-8fde-7dadf71f5695)|![Simulator Screenshot - iPhone 15 - 2024-05-13 at 16 39 08](https://github.com/pagopa/io-app/assets/5150343/7f4e905c-1a4c-4b87-9f9f-7165ff7cf3c7)|

## List of changes proposed in this pull request
- New `isNewHomeSectionEnabled` property on `persistedPreferences`, with its migration
- `DeveloperModeSection` component has a new switch to control such property value
- `TabNavigator` spawns either the Legacy or the new `MessagesHomeScreen`
- Old `MessagesHomeScreen` renamed to `LegacyMessagesHomeScreen`

## How to test
Using the io-dev-api-server, check that both the new and the old messages home screen is displayed, when the related toggle is changed in the debug profile options.
